File: pfm-sdk/src/main/java/com/tink/pfmsdk/security/DefaultRecoveryHandler.java
package com.tink.pfmsdk.security;

import android.content.SharedPreferences;
import java.security.KeyStore;
import java.security.KeyStoreException;
import java.util.List;

public class DefaultRecoveryHandler extends RecoveryHandler {

	@Override
	protected boolean recover(Exception e, KeyStore keyStore, List<String> keyAliases,
		SharedPreferences preferences) {
		e.printStackTrace();

		try {
			clearKeyStore(keyStore, keyAliases);
			clearPreferences(preferences);
			return true;
		} catch (KeyStoreException e1) {
			e1.printStackTrace();
		}

		return false;
	}
}

File: pfm-ui/src/main/java/com/tink/pfmui/security/DefaultRecoveryHandler.kt
package com.tink.pfmui.security

import android.content.SharedPreferences
import java.security.KeyStore
import java.security.KeyStoreException

internal class DefaultRecoveryHandler : RecoveryHandler() {
    override fun recover(
        e: Exception,
        keyStore: KeyStore,
        keyAliases: List<String>,
        preferences: SharedPreferences
    ): Boolean {
        e.printStackTrace()
        try {
            clearKeyStore(keyStore, keyAliases)
            clearPreferences(preferences)
            return true
        } catch (e1: KeyStoreException) {
            e1.printStackTrace()
        }
        return false
    }
}